src/System.IO.Pipes/src/System/IO/Pipes/PipeStream.Unix.cs

```diff
@@ -255,6 +226,59 @@ public virtual PipeTransmissionMode ReadMode
         // ---- PAL layer ends here ----
         // -----------------------------
 
+        private static string s_pipeDirectoryPath;
+
+        private static string GetPipeDirectoryPath()
+        {
+            string path = s_pipeDirectoryPath;
+            if (path == null)
```

Comment:
So in some cases we will have path == null but the directory will actually exist because it was created (but not cleaned up) by an earlier running instance. In this case, we will go through three calls to CreateDirectory that that return Interop.Error.EEXIST to ensure the directory exists.

Is it worthwhile to instead start with a check to see if the directory already exists a la Directory.Exists (using GetFileInformationFromPath for Unix)?